File: csv-query/typescript/tests/QueryBuilder.ts
import { parseCsv } from '../src/CsvParser.js';
import { Query } from '../src/Query.js';

const DefaultCsv = [
  'name,age,city,salary',
  'Alice,35,London,75000',
  'Bob,28,Paris,55000',
  'Charlie,42,London,90000',
  'Diana,31,Berlin,65000',
  'Eve,28,Paris,60000',
].join('\n');

export class QueryBuilder {
  private csv = DefaultCsv;

  withCsv(csv: string): this { this.csv = csv; return this; }

  build(): Query {
    const table = parseCsv(this.csv);
    return new Query(table);
  }
}

File: csv-query/python/tests/query_builder.py
from __future__ import annotations

from csv_query import Query, parse_csv

DEFAULT_CSV = (
    "name,age,city,salary\n"
    "Alice,35,London,75000\n"
    "Bob,28,Paris,55000\n"
    "Charlie,42,London,90000\n"
    "Diana,31,Berlin,65000\n"
    "Eve,28,Paris,60000"
)


class QueryBuilder:
    def __init__(self) -> None:
        self._csv = DEFAULT_CSV

    def with_csv(self, csv: str) -> QueryBuilder:
        self._csv = csv
        return self

    def build(self) -> Query:
        table = parse_csv(self._csv)
        return Query(table)
